### ISSUE TITLE
How to train the ResNext model on the Cityscapes?

### ISSUE THREAD

**Issue by PkuRainBow** [mapillary/inplace_abn#23]
I used to train the PSP with the original ResNet-101(7x7 convolution) and achieve 76.8 mIou on the val set of Cityscapes. Thus I want to replace the ResNet-101 with the ResNext-101(the imagenet pretrained model you provide) and train it with the same parameters, but the performance seems very poor(66.4). Do you use the COCO to finetune the ResNext model?

Could you give me some tips to tune the hyperparameters for the ResNext-101 compared with ResNet-101?

Besides, I am wondering whether have you trained the modified ResNet-101(replace the 7X7 conv with three 3x3 conv) model on the ImageNet? If so, it would be great if you could share me this model.

**Comment by Paranoidv** [comment]
@PkuRainBow 
Hi, how to train PSPNet with with the original ResNet-101(7x7 convolution) and achieve 76.8 mIou ?? I have train PSPNet once, but just get miou about 65.0, can you share me the code and model ? Thank you !!

**Comment by PkuRainBow** [comment]
@Paranoidv We plan to release the code in the future days. The current times is not convenient as we are preparing for a paper.

**Comment by Paranoidv** [comment]
@PkuRainBow 
Thanks for your reply ,I understand. en, I want to ask that when you train PSPNet, have you use the pretrained ResNet-101 or not, and what the batch_size is ?

**Comment by PkuRainBow** [comment]
@Paranoidv I used the COCO pretrained model. Batch size is set 8. LR=7e-3, weight decay=5e-4,  all the other settings follow the Deeplabv3. We only train the model for 4w iterations. More iterations could lead to slighly better performance.

**Comment by PkuRainBow** [comment]
@Paranoidv [](https://github.com/CSAILVision/semantic-segmentation-pytorch) This repo seems a good start point~

**Comment by Paranoidv** [comment]
@PkuRainBow 
Very appreciate! I have one GPU, so, I set the batch_size=2, I will try with you advice !

**Comment by Paranoidv** [comment]
@PkuRainBow 
Another question, the original thesis about PSPNet says that miou is 81.2 on Cityscapes, have you get the same or better miou??

**Comment by PkuRainBow** [comment]
@Paranoidv I recomend to to apply more resources. We run all of experiments with 4X GPUs.  We submit and achieved 79.7(w/o coarse data) with our own method. The original PSP can only achieve around 77~. We still can not reproduce their 81.2 performance.

**Comment by Paranoidv** [comment]
@PkuRainBow 
Thanks !! I will try, big batch_size could get better miou？ there may be connection

**Comment by tt-leader** [comment]
@PkuRainBow Hi, I'm reproducing deeplab v3+ on cityscapes, and archive 79.3% at test split without coarse annotations. Did you use coarse data? And how will you use coarse data to increase your performance?

**Comment by PkuRainBow** [comment]
@tt-leader  Do you use ms + flip? Besides, do you use validation set for training?

**Comment by ducksoup** [comment]
I'm closing this issue since it doesn't appear to be related with the InPlace-ABN code.
